source/data-modeling/persistence-configuration.txt

```diff
@@ -0,0 +1,296 @@
+When you call ``with()`` on a model class or instance, the class or instance is 
+yielded to the block. The temporary persistence context applies to both queries 
+and writes.
+
+By default, {+odm+} stores documents for the ``Band`` class in a collection called
+``bands``. The following code example uses the ``with()`` method to temporarily
+use a different client, database, and collection to perform operations on the 
+``Band`` class's documents:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start with() example
+   :end-before: end with() example
+
+.. note:: Define Clients
+
+   In the preceding example, you must define the ``tertiary`` cluster under
+   ``clients`` in your ``mongoid.yml`` file.
+
+Ensuring Consistency in Contexts
+++++++++++++++++++++++++++++++++
+
+If you save a document in a temporary persistence context, it does not exist
+in the default persistence context if you do not save it there as well. 
+For example, if you insert a ``Band`` model document into the ``artists`` 
+collection, the same document will not be inserted into the ``bands`` collection.
+Ensure that you perform operations on the same persistence context for data
+consistency. The following example uses ``with()`` to insert a document into
+a custom collection and to update that document in the correct collection.
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start consistency example
+   :end-before: end consistency example
+
+.. important:: Block Scope
+   
+   You must call the ``with()`` method with a block.
+   This is because {+odm+} uses the options you pass to the method to create
+   a new client in the background. A block defines the scope of this client
+   so it can be closed and its resources freed.
+
+You can also perform read and write operations in the context of a ``with()``
+method. The following example uses the ``with()`` method to call ``Band.count``
+on the secondary node of a replica set:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start read configuration
+   :end-before: end read configuration
+
+Global Persistence Contexts
++++++++++++++++++++++++++++
+
+In previous examples in this section, you can only change persistence context in
+the scope of a block. {+odm+} can globally define the persistence
+configuration to perform all operations on a particular database or client. 
+This lets you change the persistence context for all operations at runtime
+without repreatedly calling the ``with()`` method.
+
+You can use the following methods to globally define the persistence context
+in your program:
+
+- ``{+odm+}.override_client``: Perform all operations on the specified client.
+
+- ``{+odm+}.override_database``: Perform all operations on the specified
+  database.
```

Comment:
S: 'Perform' reads as imperative instead of indicative

```suggestion
- ``{+odm+}.override_client``: {+odm+} performs all operations on the specified client.
- ``{+odm+}.override_database``: {+odm+} performs all operations on the specified
  database.
```